subprojects/testing-native/src/test/groovy/org/gradle/nativeplatform/test/xctest/internal/XCTestSelectionTest.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest.internal
+
+import org.gradle.testing.internal.util.Specification
+import spock.lang.Subject
+
+@Subject(XCTestSelection)
+class XCTestSelectionTest extends Specification {
```

Comment:
Could do with some more test coverage for broken patterns, eg leading '.' or trailing '.' or multiple '.' with nothing between them. Also patterns that use the internal representation for tests, eg `/a.b.c` or `a.b/c.d`.

Reply:
What should be the expected result of `a.b/c` right now, this will work correctly as it matches what we are going to pass to XCTest. Should we explicitly prevent the `/` character forcing users to use `a.b.c` instead?